src/main/java/net/fabricmc/loom/processors/aw/AccessWidener.java

```diff
@@ -40,13 +40,17 @@
 import net.fabricmc.mappings.EntryTriple;
 
 public class AccessWidener {
-	public String namespace;
-	public Map<String, Access> classAccess = new HashMap<>();
-	public Map<EntryTriple, Access> methodAccess = new HashMap<>();
-	public Map<EntryTriple, Access> fieldAccess = new HashMap<>();
-	private Set<String> classes = new LinkedHashSet<>();
+	public final String namespace;
+	public final Map<String, Access> classAccess = new HashMap<>();
+	public final Map<EntryTriple, Access> methodAccess = new HashMap<>();
+	public final Map<EntryTriple, Access> fieldAccess = new HashMap<>();
+	private final Set<String> classes = new LinkedHashSet<>();
+
+	public AccessWidener(String namespace) {
+		this.namespace = namespace;
+	}
 
-	public void read(BufferedReader reader) throws IOException {
+	public AccessWidener(BufferedReader reader) throws IOException {
```

Comment:
Can we make this a static method that can return a dummy widener if an io exception arises?

Reply:
Wouldn’t that prevent your project from compiling?